Testnet/NonFungibleToken/NonFungibleToken/NonFungibleToken.cs

```diff
@@ -420,53 +433,48 @@ private void LogApprovalForAll(Address owner, Address operatorAddress, bool appr
     /// Guarantees that the <see cref="Message.Sender"/> is an owner or operator of the given NFT.
     /// </summary>
     /// <param name="tokenId">ID of the NFT to validate.</param>
-    private void CanOperate(UInt256 tokenId)
+    private void CanOperate(Address tokenOwner)
     {
-        Address tokenOwner = GetIdToOwner(tokenId);
         Assert(tokenOwner == Message.Sender || GetOwnerToOperator(tokenOwner, Message.Sender));
     }
 
     /// <summary>
     /// Guarantees that the msg.sender is allowed to transfer NFT.
     /// </summary>
     /// <param name="tokenId">ID of the NFT to transfer.</param>
-    private void CanTransfer(UInt256 tokenId)
+    private void CanTransfer(Address tokenOwner, UInt256 tokenId)
     {
-        Address tokenOwner = GetIdToOwner(tokenId);
         Assert(
           tokenOwner == Message.Sender
           || GetIdToApproval(tokenId) == Message.Sender
           || GetOwnerToOperator(tokenOwner, Message.Sender)
         );
     }
 
-    /// <summary>
-    /// Guarantees that tokenId is a valid Token.
-    /// </summary>
-    /// <param name="tokenId">ID of the NFT to validate.</param>
-    private void ValidateToken(UInt256 tokenId)
+    public void SetPendingOwner(Address newOwner)
     {
-        Address tokenOwner = GetIdToOwner(tokenId);
-        EnsureAddressIsNotEmpty(tokenOwner);
+        EnsureOwnerOnly();
+        PendingOwner = newOwner;
+
+        Log(new OwnershipTransferRequestedLog { CurrentOwner = Owner, PendingOwner = newOwner });
     }
 
-    /// <summary>
-    /// Sets the contract owner who can mint/burn
-    /// </summary>
-    /// <param name="owner"></param>
-    public void TransferOwnership(Address owner)
+    public void ClaimOwnership()
```

Comment:
Method looks good but consider adding the comment without the owner parameter since its removed.